Update talent plan 2.0 (#338)
pingcap/talent-plan: README.md

# Welcome to learn Talent Plan Courses!

![Talent Plan Logo](media/talent-plan-logo.png)

Talent Plan is an open source training program initiated by PingCAP. It aims to create or combine some open source learning materials for people interested in open source, distributed systems, Rust, Golang, and other infrastructure knowledges. As such, a series of courses focused on open source collaboration, rust programming, distributed database and systems are provided.

> Note:

Reviewer: ```suggestion
> Note:
```

- [TP 101: Introduction to open source software](courses/tp101-intro-to-oss.md)
- [TP 102: How to use Git and GitHub](courses/tp102-how-to-use-git-github.md)
- [TP 103: Build a welcoming community](courses/tp103-open-source-community.md)

### Series 2: Rust Programming

Two courses are included in this series, which are:

- [TP 201: Practical Networked Applications in Rust](rust/README.md). A series of projects that incrementally develop a single Rust project from the ground up into a high-performance, networked, parallel and asynchronous key/value store. Along the way various real-world and practical Rust development subject matter are explored and discussed.

- [TP 202: Distributed Systems in Rust](dss/README.md). Adapted from the [MIT 6.824](http://nil.csail.mit.edu/6.824/2017/index.html) distributed systems coursework, this course focuses on implementing important distributed algorithms, including the [Raft](https://raft.github.io/) consensus algorithm, and the [Percolator](https://storage.googleapis.com/pub-tools-public-publication-data/pdf/36726.pdf) distributed transaction protocol.

### Series 3: Distributed Database 

Two courses are included in this series, which are:

> - TP 301: TinyKV, a distributed key value database in Go(WIP) 

Reviewer: should we link them to the repository of `tinysql` and `tinykv`?

Author: It seems to work, what do you think? @Connor1996 @winoros

Reviewer 2: so, we do not need to move it into talent-plan repo?

Author: yes, we can link it to the repository in `pingcap-incubator` org directly.

> - TP 301: TinyKV, a distributed key value database in Go(WIP) 
> - TP 302: TinySQL, a distributed relational database in Go(WIP)

Author: ```suggestion
- [TP 301: TinyKV, a distributed key value database in Go](https://github.com/pingcap-incubator/tinykv) 
- [TP 302: TinySQL, a distributed relational database in Go](https://github.com/pingcap-incubator/tinysql)
```

Author: Maybe we can link `tinysql` and `tinykv`to the repository in `pingcap-incubator` org directly, still we'd better add some brief description of these two courses in `REAMDE.md`. What do you think?

@zz-jason @Connor1996 @winoros

Reviewer 2: oh, that would be better

Author: ok, then I'll commit this suggestion. Could you please add a brief description to tinysql and tinykv courses?

@Connor1996 @winoros

### Series 4: Deep Dive into TiDB Ecosystems 

> - Deep Dive into TiDB(WIP)
> - Deep Dive into TiKV(WIP)

Reviewer: ```suggestion
- TP 401: Deep Dive into TiDB(WIP)
- TP 402: Deep Dive into TiKV(WIP)
```
